### ISSUE TITLE
tab type auto works only after reopening the file

### ISSUE THREAD

**Issue by sandrodz** [atom/atom#11768]
### Description

tab type auto detects (soft|hard) correctly only after closing file and re-opening it.

### Steps to Reproduce

1. Open file that contains soft tabs.

![screen shot 2016-05-16 at 4 15 45 pm](https://cloud.githubusercontent.com/assets/8479569/15289180/798d09d4-1b81-11e6-883e-d63599453570.png)

2. Change soft tabs to hard tabs. (I use tabify)
3. save
4. Create new line, click tab

**Expected behavior:** hard tab created on new line
![screen shot 2016-05-16 at 4 15 11 pm](https://cloud.githubusercontent.com/assets/8479569/15289209/a6482b34-1b81-11e6-98d7-e6443046bb92.png)

**Actual behavior:** soft tab will be created
![screen shot 2016-05-16 at 4 14 53 pm](https://cloud.githubusercontent.com/assets/8479569/15289215/b353eaa2-1b81-11e6-8436-6ea75795392b.png)

after closing the file, and opening again, and redoing 4th step, it correctly creates hard tab.

### Versions

OSX 10.11.4

Atom : 1.7.3
Electron: 0.36.8
Chrome : 47.0.2526.110
Node : 5.1.1

apm 1.9.2
npm 2.13.3
node 0.10.40
python 2.7.10
git 2.7.4

**Comment by 50Wliu** [comment]
This seems like a duplicate of #4054, so feel free to subscribe there for updates.
